Source/Core/AxisAlignedBoundingBox.js

```diff
@@ -191,6 +193,27 @@ define([
         return Intersect.INTERSECTING;
     };
 
+    /**
+     * If the given position is not already within the box, projects the given position onto the box
+     * @param {Cartesian3} position The position being projected onto this AxisAlignedBoundingBox.
+     * @returns {Cartesian3} A projected version of the inputted position if it was not originally within the AxisAlignedBoundingBox.
+     */
+    AxisAlignedBoundingBox.prototype.projectedPoint = function(position, result) {
```

Comment:
All of these new functions will need specs too.

Reply:
Since the function will not modify the position if already inside the box, is `project` the best terminology? I think we may have touched on the naming before, so I'm sorry if we already came to a conclusion.